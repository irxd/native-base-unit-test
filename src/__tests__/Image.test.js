import React from 'react';
import { render } from '@testing-library/react-native';
import {
  NativeBaseProvider, Image
} from 'native-base';

describe('<Image />', () => {
  it('should render image component properly', () => {
    const { queryByTestId } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <Image
          source={{ uri: "https://wallpaperaccess.com/full/317501.jpg"}}
          alt="Alternate Text"
          size="xl"
          testID="my-image"
        />
      </NativeBaseProvider>
    );

    expect(queryByTestId('my-image')).not.toBeNull()
  });

  it('should render image props properly', () => {
    const { queryByTestId, toJSON } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <Image
          source={{ uri: "https://wallpaperaccess.com/full/317501.jpg"}}
          alt="Alternate Text"
          width={500}
          height={500}
          testID="my-image"
        />
      </NativeBaseProvider>
    );
    
    expect(queryByTestId('my-image').props.source.uri).toMatch("https://wallpaperaccess.com/full/317501.jpg");
    expect(queryByTestId('my-image').props.style.width).toEqual(500);
    expect(queryByTestId('my-image').props.style.height).toEqual(500);
    expect(toJSON()).toMatchSnapshot();
  });
});
