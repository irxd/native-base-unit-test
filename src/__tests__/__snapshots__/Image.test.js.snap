// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render image props properly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Image
    accessibilityLabel="Alternate Text"
    alt="Alternate Text"
    dataSet={Object {}}
    onError={[Function]}
    source={
      Object {
        "uri": "https://wallpaperaccess.com/full/317501.jpg",
      }
    }
    style={
      Object {
        "height": 500,
        "maxWidth": "100%",
        "width": 500,
      }
    }
    testID="my-image"
  />
</RNCSafeAreaProvider>
`;
